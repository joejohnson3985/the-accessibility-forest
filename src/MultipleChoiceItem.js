import React, { Component } from 'react';
import './MultipleChoiceItem.scss';

class MultipleChoiceItem extends Component {
  constructor(props) {
    super(props);
    this.state = {
    }
  }

  handleClick = () => {
    this.props.handleAnswer(this.props.choice)
  }

  render() {
    return (
      <div>
        <div className='choice-btn' role='button' onClick= {this.handleClick}>{this.props.choice}</div>
      </div>
    )
  }
}

export default MultipleChoiceItem;