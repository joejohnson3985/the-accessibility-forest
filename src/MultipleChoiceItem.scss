@import './_variables.scss';

.choice-btn {
  @include secondary-button;
  margin: 10px 0;

  &:hover {
    background-color: $secondary-color;
  }

  &:active {
    @include active-btn;
  }
}

