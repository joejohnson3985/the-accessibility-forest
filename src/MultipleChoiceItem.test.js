
import React from 'react';
import MultipleChoiceItem from './MultipleChoiceItem';
import { shallow } from 'enzyme';

const mockFunc = jest.fn();
const mockChoice = "<address>"
const mockKey = "testkey"

describe('MultipleChoiceItem', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
        <MultipleChoiceItem
          choice={mockChoice} 
          key={mockKey} 
          handleAnswer={mockFunc}
        />
      )
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('Should fire handleClick on click', () => {
    wrapper.find('.choice-btn').simulate('click', {preventDefault: () => {} })
    expect(mockFunc).toBeCalled();
  })
})