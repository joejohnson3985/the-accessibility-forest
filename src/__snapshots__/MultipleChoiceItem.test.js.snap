// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultipleChoiceItem Should match the snapshot 1`] = `
<div>
  <div
    className="choice-btn"
    onClick={[Function]}
    role="button"
  >
    &lt;address&gt;
  </div>
</div>
`;
